rules/DowngradePhp71/Rector/StaticCall/DowngradeClosureFromCallableRector.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp71\Rector\StaticCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\ClosureUse;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\Return_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Naming\Naming\VariableNaming;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\PostRector\Collector\NodesToAddCollector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/closurefromcallable
+ *
+ * @see \Rector\Tests\DowngradePhp71\Rector\StaticCall\DowngradeClosureFromCallableRector\DowngradeClosureFromCallableRectorTest
+ */
+final class DowngradeClosureFromCallableRector extends AbstractRector
+{
+    public function __construct(
+        private NodesToAddCollector $nodesToAddCollector,
+        private VariableNaming $variableNaming
+    ) {
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [StaticCall::class];
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Converts Closure::fromCallable() to compatible alternative.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+\Closure::fromCallable('callable');
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$callable = 'callable';
+function () use ($callable) {
+    return $callable(...func_get_args());
+};
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $tempVariableName = $this->variableNaming->createCountedValueName(
+            'callable',
+            $node->getAttribute(AttributeKey::SCOPE)
+        );
+
+        $tempVariable = new Variable($tempVariableName);
+
+        $assignment = new Expression(
+            new Assign(
+                $tempVariable,
+                $node->args[0]->value
+            )
+        );
+
+        $currentStatement = $node->getAttribute(AttributeKey::CURRENT_STATEMENT);
+        $this->nodesToAddCollector->addNodeBeforeNode($assignment, $currentStatement);
```

Comment:
```suggestion
        $this->addNodeBeforeNode($assignment, $currentStatement);
```

Reply:
Actually, with `$this->addNodeBeforeNode($assignment, $currentStatement);` I am getting a deprecated warning. Are you sure?

Reply:
Other rules still use it, I think that need to be in separate PR like "[PostRector] Update to use NodesToAddCollector in rector rules" PR so that ease to track the change, since `nodesToAddCollector` property is now protected in `AbstractRector` so a search - replace PR should can do that under `rules`.

Reply:
Check!